presto-jdbc/src/main/java/io/prestosql/jdbc/PrestoPreparedStatement.java

```diff
@@ -67,6 +75,9 @@
         extends PrestoStatement
         implements PreparedStatement
 {
+    private static final Pattern topLevelTypeRegexPattern = Pattern.compile("^(.+?)\\((.+)\\)$");
+    private static final Pattern timestampTimeZonePrecisionRegexPattern = Pattern.compile("^timestamp\\((\\d+)\\) with time zone$");
```

Comment:
We also need to handle `time with time zone`. 

Technically, `timestamp` (and `time`) could be written as `timestamp without time zone`, but we don't need to handle it since the server won't return that optional suffix.

Reply:
Thanks! Handled `time(n) with time zone` here. Please correct me if I am wrong